src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/AzureEventHubsCmdletBase.cs

```diff
@@ -174,4 +176,33 @@ public Hashtable ConvertTagsToHashtable(IDictionary<string, string> tags)
 
         #endregion
     }
+    public class LocalResourceIdentifier : ResourceIdentifier
+    {
+        public LocalResourceIdentifier(string strResourceID) : base(strResourceID)
+        {
+            string[] tokens = base.ParentResource.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+
+            // for upto 3 Parents 
+            switch (tokens.Length)
+            {
+                case 2:
+                    ParentResource = tokens[1];
```

Comment:
@v-Ajnava [same comment I made in the ServiceBus PR](https://github.com/Azure/azure-powershell/pull/6732#discussion_r204581707) -- wouldn't this `tokens[1]` variable map to the subscription id?

Reply:
Yes, its a resource id, but the tokens here will be the ParentResource property of ResourceIdentifier (Microsoft.Azure.Management.Internal.Resources.Utilities.Models) class. Tokens will have only parent resources.